src/main.py

```diff
@@ -185,49 +188,79 @@ def main(args):
 
     with strategy.scope():
         print("######### Preparing model #########")
-        model = Wav2Vec2ForCTCTrainer.from_pretrained(
-            args.model_id,
-            jit_compile=jit_compile,
-            input_shape=model_input_shape,
-            apply_spec_augment=args.apply_spec_augment,
-            survival_prob=args.survival_prob,
-        )
-
-        # during fine-tuning, we need to freeze the feature extraction layer from the pre-trained weights
-        model.freeze_feature_extractor()
+        model = build_model(args)
 
         # `division_factor` should be passed else loss will be summed
         # it will help us in distributed training over several processes
-        loss_fn = CTCLoss(
+        loss = CTCLoss(
             model.config, model_input_shape, division_factor=global_batch_size
         )
 
-        optimizer = tf.keras.optimizers.Adam(learning_rate=args.lr)
-        model.compile(optimizer, loss_fn)
-
-    model.summary()
-    print("######### Initiating training #########")
-    model.fit(
-        tr_dataset,
-        validation_data=val_dataset,
-        epochs=args.max_epochs,
-        callbacks=fetch_callbacks(args),
-        verbose="auto",
-    )
-
-    print("\n######### Preparing for evaluation #########")
+        ######################### STAGE-1 #########################
+
+        print("######## FREEZING ########")
```

Comment:
A better and more informative comment might be: 

> Freezing the backbone.

Or the section of the network we are freezing like stem, body, trunk, etc.